EVA-3763 - Add nextflow configuration documentation. (#72)
EBIvariation/eva-sub-cli: README.md

If you are working with large VCF files and find that validation takes a very long time, you can add the
argument `--shallow` to the command, which will validate only the first 10,000 lines in each VCF. Note that running
shallow validation will **not** be sufficient for actual submission.

> ## Leveraging Nextflow to parallelize the validation process  

Reviewer: Do you think this should go in the main readme or in a separate "advanced usage" doc?  It feels technical but on the other hand it's actually pretty important...

Author: I hesitated but ultimately felt like it was too short  for its own document. And since it will be used by everyone regardless of if they want to use advance feature I think we'll need to have some of it in the main readme anyway.

> When running natively (i.e. not using docker), eva-sub-cli will use nextflow to run all the validation steps. When no options are provided nextflow will run as many tasks as there are available CPU on the machine executing it. To modify how many tasks can start and how nextflow will process each one, you can provide a nextflow configuration file in several ways:

Reviewer: ```suggestion
When running natively (i.e. not using Docker), eva-sub-cli will use a workflow management system called Nextflow to run all the validation steps. When no options are provided, Nextflow will run as many tasks as there are available CPUs on the machine executing it. To modify how many tasks can start and how Nextflow will process each one, you can provide a Nextflow configuration file in several ways.
```
* I think Nextflow and Docker should be capitalized throughout...
* Somewhere we should say briefly what Nextflow is as many people probably won't know, either here or even earlier when we define the dependencies.

From the command line you can use `--nextflow_config <path>` to specify the nextflow config file you want to apply. The configuration can also be picked up from other places directly by nextflow. Please refer to [the nextflow documentation](https://www.nextflow.io/docs/latest/config.html) for more details.

### Basic nextflow configuration.

> There are many options to configure nextflow so we will not provide them all. Please refer to [the documentation](https://www.nextflow.io/docs/latest/reference/config.html) for advanced feature 

Reviewer: ```suggestion
There are many options to configure nextflow so we will not provide them all. Please refer to [the documentation](https://www.nextflow.io/docs/latest/reference/config.html) for advanced features.
```

Below is a very basic nextflow configuration file that will request 2 cpus for each process, essentially limiting the number of process to half the number of available CPUs 
```
process {
    executor="local"
    cpus=2
}
```
In this configuration, all the process will be running on the same machine where eva-sub-cli was started.
```
(Local machine)
eva-sub-cli
  |_ nextflow
      |_ task1
      |_ task2
```

If you have access to High Performance Compute environment, nextflow supports the main resource managers such as [SLURM](https://www.nextflow.io/docs/latest/executor.html#slurm), [SGE](https://www.nextflow.io/docs/latest/executor.html#sge), [LSF](https://www.nextflow.io/docs/latest/executor.html#lsf) and others.
```
process {
    executor="slurm"
    queue="my_production_queue"
}
```

> In this configuration, the subtasks will be performed in other machine as specified by your SLURM resource manager.

Reviewer: ```suggestion
In this configuration, the subtasks will be performed in other machines as specified by your SLURM resource manager.
```
